src/txn/txn.c

```diff
@@ -127,7 +127,7 @@ __wt_txn_release_snapshot(WT_SESSION_IMPL *session)
     WT_ASSERT_OPTIONAL(session,
       txn_shared->pinned_id == WT_TXN_NONE || session->txn->isolation == WT_ISO_READ_UNCOMMITTED ||
         !__wt_txn_visible_all(session, txn_shared->pinned_id, WT_TS_NONE),
-      "Releasing snapshot that is not globally visible");
+      "A transactions pinned id cannot become globally visible before the snapshot is released");
```

Comment:
I know I suggested this working but I'd update the grammar from "the snapshot" to "its snapshot".